src/theory/quantifiers/ematching/pattern_term_selector.h

```diff
@@ -46,7 +46,8 @@ class PatternTermSelector
    * also returning f(f(x)). This is default true since it helps in practice
    * to filter trigger instances.
    */
-  PatternTermSelector(Node q,
+  PatternTermSelector(const Options& opts,
```

Comment:
You need t add `opts` to the docs above. 

As a general comment it is a bit unfortunate to have to pass the options to all the static methods bellow. These methods are used in very few places, so maybe you could change the design to have these places use an instance of this object? Or have it leave in one of the "global" databases?